source/client-side-encryption/client-side-encryption.md

```diff
@@ -0,0 +1,2475 @@
+
+The following pseudo-code describes the configuration behavior for the three `MongoClient` objects:
+
+```
+def getOrCreateInternalClient (client, clientOpts):
+   if client.internalClient != None:
+      return client.internalClient
+   internalClientOpts = copy(clientOpts)
+   internalClientOpts.autoEncryptionOpts = None
+   internalClientOpts.minPoolSize = 0
+   client.internalClient = MongoClient (internalClientOpts)
+   return client.internalClient
+
+def configureAutoEncryptionClients (client, clientOpts):
+   if clientOpts.autoEncryptionOpts.keyVaultClient != None:
+      client.keyVaultClient = clientOpts.autoEncryptionOpts.keyVaultClient
+   elif clientOpts.maxPoolSize == 0:
+      client.keyVaultClient = client
+   else:
+      client.keyVaultClient = getOrCreateInternalClient (client, clientOpts)
+
+   if clientOpts.autoEncryptionOpts.bypassAutomaticEncryption:
+      client.metadataClient = None
+   elif clientOpts.maxPoolSize == 0:
+      client.metadataClient = client
+   else:
+      client.metadataClient = getOrCreateInternalClient (client, clientOpts)
+```
+
+Configuring the internal `MongoClient` MUST match the parent `MongoClient`, except `minPoolSize` is set to `0` and
+`AutoEncryptionOpts` is omitted. This includes copying the options and host information from the URI, and other non-URI
+configuration (monitoring callbacks, stable API, etc.).
+
+Drivers MUST document that an additional `MongoClient` may be created, using the following as a template:
+
+> If a `MongoClient` with a limited connection pool size (i.e a non-zero `maxPoolSize`) is configured with
+> `AutoEncryptionOpts`, a separate internal `MongoClient` is created if any of the following are true:
+>
+> - `AutoEncryptionOpts.keyVaultClient` is not passed.
+> - `AutoEncryptionOpts.bypassAutomaticEncryption` is `false`.
+>
+> If an internal `MongoClient` is created, it is configured with the same options as the parent `MongoClient` except
+> `minPoolSize` is set to `0` and `AutoEncryptionOpts` is omitted.
+
+See
+[What's the deal with metadataClient, keyVaultClient, and the internal client?](#whats-the-deal-with-metadataclient-keyvaultclient-and-the-internal-client)
+
+<div id="GCPKMSOptions">
+<div id="AWSKMSOptions">
+<div id="KMSProvider">
+<div id="KMSProviders">
+<div id="AzureAccessToken">
+<div id="kmsproviders">
+
+#### kmsProviders
+
+The `kmsProviders` property may be specified on [ClientEncryptionOpts](#ClientEncryptionOpts) or
+[AutoEncryptionOpts](#AutoEncryptionOpts). Multiple KMS providers may be specified, each using a specific property on
+the [KMSProviders](#kmsproviders) object. The options differ for each KMS provider type. The "local" KMS provider type
+is configured with master key material. The external providers are configured with credentials to authenticate.
+
+Throughout this document, the KMS provider is annotated as - Ts:`KMSProvider`, but this name is for *exposition only*:
```

Comment:
```
Ts:`KMSProvider`
```

Is that supposed to literally display "Ts:`KMSProvider`" in the rendered output?

Reply:
It was an artifact, fixed.